sdk/ai/Azure.AI.Projects/src/Custom/AIProjectClient.cs

```diff
@@ -2,23 +2,61 @@
 // Licensed under the MIT License.
 
 using System;
-using System.Linq;
-using Azure.AI.Inference;
+using System.ClientModel.Primitives;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Threading;
 using Azure.Core;
+using Azure.Identity;
 
 namespace Azure.AI.Projects
 {
     // Data plane generated client.
     /// <summary> The AzureAI service client. </summary>
-    public partial class AIProjectClient
+    public partial class AIProjectClient : ConnectionProvider
     {
+        /// <summary>
+        /// Protects the <see cref="Connections"/> collection from concurrent access. Separated from <see cref="_connectionCacheLock"/> to reduce contention.
+        /// </summary>
+        private readonly ReaderWriterLockSlim _connectionsLock = new ReaderWriterLockSlim(LockRecursionPolicy.NoRecursion);
+
+        /// <summary>
+        /// Protects the <see cref="_connectionCache"/> dictionary from concurrent access. Separated from <see cref="_connectionsLock"/> to improve concurrency in read-heavy scenarios.
+        /// </summary>
+        private readonly ReaderWriterLockSlim _connectionCacheLock = new ReaderWriterLockSlim(LockRecursionPolicy.NoRecursion);
+
+        /// <summary>
+        /// Subclient connections.
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public ConnectionCollection Connections { get; } = [];
+
+        private readonly ConnectionsClient _connectionsClient;
+
+        private readonly Dictionary<ConnectionType, ConnectionResponse> _connectionCache = new();
+
         /// <summary> Initializes a new instance of AzureAIClient. </summary>
         /// <param name="connectionString">The Azure AI Foundry project connection string, in the form `endpoint;subscription_id;resource_group_name;project_name`.</param>
         /// <param name="credential"> A credential used to authenticate to an Azure Service. </param>
         /// <exception cref="ArgumentNullException"> <paramref name="connectionString"/> is null. </exception>
         /// <exception cref="ArgumentException"> <paramref name="connectionString"/> </exception>
-        public AIProjectClient(string connectionString, TokenCredential credential) : this(connectionString, credential, new AIProjectClientOptions())
+        public AIProjectClient(string connectionString, TokenCredential credential = null) : this(connectionString, BuildCredential(credential), new AIProjectClientOptions())
         {
+            _connectionsClient = new ConnectionsClient(connectionString, _tokenCredential);
+        }
+
+        private static TokenCredential BuildCredential(TokenCredential credential)
+        {
+            if (credential != null)
+            {
+                return credential;
+            }
+
+            string clientId = Environment.GetEnvironmentVariable("CLOUDMACHINE_MANAGED_IDENTITY_CLIENT_ID");
+
+            return !string.IsNullOrEmpty(clientId)
```

Comment:
@schaabs, this is the first client library that defaults credentials (and so takes dependency on Identity). Do you think it's OK?

@dargilco, @jhakulin